packages/core/src/audio/AudioClip.ts

```diff
@@ -0,0 +1,52 @@
+import { Engine } from "../Engine";
+import { ReferResource } from "../asset/ReferResource";
+
+/**
+ * Audio Clip
+ */
+export class AudioClip extends ReferResource {
+  /** the name of clip */
+  name: string;
+
+  private _audioBuffer: AudioBuffer;
+
+  /**
+   * the number of discrete audio channels
+   */
+  get channels(): Readonly<number> {
+    return this._audioBuffer.numberOfChannels;
+  }
+
+  /**
+   * the sample rate, in samples per second
+   */
+  get sampleRate(): Readonly<number> {
+    return this._audioBuffer.sampleRate;
+  }
+
+  /**
+   * the duration, in seconds
+   */
+  get duration(): Readonly<number> {
+    return this._audioBuffer.duration;
+  }
+
+  /**
+   * get the clip's audio buffer
+   */
+  getData(): AudioBuffer {
```

Comment:
Avoid web native interfaces as much as possible

Reply:
I think getData and setData is a must